src/main/php/Handler/LokiHandler.php

```diff
@@ -90,17 +94,20 @@ private function sendPacket(array $packet): void
         }
 
         if (false !== $this->connection) {
-            $curlOptions = [
-                CURLOPT_CONNECTTIMEOUT_MS => 100,
-                CURLOPT_TIMEOUT_MS => 200,
-                CURLOPT_CUSTOMREQUEST => 'POST',
-                CURLOPT_RETURNTRANSFER => true,
-                CURLOPT_POSTFIELDS => $payload,
-                CURLOPT_HTTPHEADER => [
-                    'Content-Type: application/json',
-                    'Content-Length: ' . strlen($payload),
+            $curlOptions = array_replace(
+                [
+                    CURLOPT_CONNECTTIMEOUT_MS => 100,
+                    CURLOPT_TIMEOUT_MS => 200,
+                    CURLOPT_CUSTOMREQUEST => 'POST',
+                    CURLOPT_RETURNTRANSFER => true,
+                    CURLOPT_POSTFIELDS => $payload,
+                    CURLOPT_HTTPHEADER => [
+                        'Content-Type: application/json',
+                        'Content-Length: ' . strlen($payload),
+                    ],
```

Comment:
Sorry, this is my fault. I didn't realize this earlier and I haven't had time to take a look at the changes properly :   
there's an issue here, well more like a risk to be exact, but it's a big one. 

If, for whatever reason, a user was to override one of the following options, logging to Loki would no longer work properly : 
- CURLOPT_CUSTOMREQUEST
- CURLOPT_RETURNTRANSFER
- CURLOPT_POSTFIELDS
- CURLOPT_HTTPHEADER

IMHO these four should never be overridable from the configuration. 
$payload being the only part that is customized anyway based on the message sent using the logger

I think here we should first override the customizable parts with the customCurlOprtions array.
Then add the non-customizable keys to resulting array.

Note : the documentation should also reflect that some options (and which ones) are not customizable.

Reply:
I disagree on this point. If the user wants to shoot him/herself into the foot, that's not our problem.

There are a lot of other `CURLOPT_*`s which would prevent the library from logging to Loki. (`CURLOPT_POST`, `CURLOPT_PUT`, `CURLOPT_CONNECT_ONLY` etc.) But the user would have to set them explicitly themselves, that doesn't happen on accident and might have a specific purpose. Therefore I don't think that it's the library's job to prevent this from happening.

Reply:
"Therefore I don't think that it's the library's job to prevent this from happening" > That's exactly what I'm saying : we're don't want to prevent changing any CURLOPT except those that we know are necessary for the system to work properly.

If a user uses a different option that happens to break the logging system that's his problem, but knowingly giving the capacity to change an option that we know is required for it to work is not a good idea : it's opening a door to invite misconfigurations / misusages that will result in Github issues that could be avoided from the start.

Reply:
Sorry, over easter I forgot about this.

So, what do you want to do? This three-step-process of defaults, customs and then unchangeable defaults applied successively to the options array?

Reply:
I guess the easiest way would be to do this with two arrays : 
- one that contains options that can be overriden with the customCurlOptions array (using array_replace, aready done)
- one that contains the options that can never be overridden and that we'd force to the correct values to ensure that the basic setup is guaranteed to work

So I'd do it this way more or less : 
```
$options = array_replace(
    $customizableOptions,
    $customCurlOptions,
    [
        // Options that cannot be overridden
    ]
);
```
And I'd set up a list in the documentation stating which options cannot be overridden and that aside from issues caused by those options, it is up to the developer to configure the logger so that it works for his or her own purpose(s).

Reply:
Finally got around to it, custom curl headers are now filtered in the constructor. Slightly faster than doing this every time `send_packet()` is called.